animation_nodes/nodes/mesh/find_shortest_path.py

```diff
@@ -0,0 +1,108 @@
+    ("STROKE", "Stroke", "Output paths as gp strokes", 2)
+]
+
+class FindShortestPathNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_FindShortestPathNode"
+    bl_label = "Find Shortest Path"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode Type", default = "TREE",
+        items = modeItems, update = AnimationNode.refresh)
+
+    pathType: EnumProperty(name = "Path Type", default = "MESH",
+        items = pathTypeItems, update = AnimationNode.refresh)
+
+    joinMeshes: BoolProperty(name = "Join Meshes", default = True,
+        update = AnimationNode.refresh)
+
+    useSourceList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh")
+        if self.mode == "PATH":
+            self.newInput("Integer", "Source", "source")
+            self.newInput("Integer", "Target", "target", value = 1)
+        else:
+            self.newInput(VectorizedSocket("Integer", "useSourceList",
+                    ("Source", "sources"), ("Sources", "sources")))
+
+        if self.mode == "PATH":
+            self.newOutput("Integer List", "Indices", "indices")
+        else:
+            if self.pathType == "MESH":
+                if not self.joinMeshes:
+                    self.newOutput("Mesh List", "Meshes", "outMeshes")
+                else:
+                    self.newOutput("Mesh", "Mesh", "outMesh")
+            elif self.pathType == "SPLINE":
+                self.newOutput("Spline List", "Splines", "outSplines")
+            elif self.pathType == "STROKE":
+                self.newOutput("GPStroke List", "Strokes", "outStrokes")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+        layout.prop(self, "pathType", text = "")
+        if self.pathType == "MESH":
+            layout.prop(self, "joinMeshes")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "PATH":
+            return "execute_Index"
+        else:
+            return "execute_All"
+
+    def execute_Index(self, mesh, source, target):
+        if mesh is None:
+            return LongList()
+
+        if source < 0 or source >= len(mesh.vertices):
+            self.raiseErrorMessage("Some indices are out of range.")
+        if target < 0 or target >= len(mesh.vertices):
+            self.raiseErrorMessage("Some indices are out of range.")
+
+        sources = LongList.fromValue(source)
+        targets = LongList.fromValue(target)
+        if self.pathType == "MESH":
+            if self.joinMeshes:
+                return Mesh.join(*getShortestPath(mesh, sources, targets, "MESH", "PATH"))
+            else:
+                return getShortestPath(mesh, sources, targets, "MESH", "PATH")
+        elif self.pathType == "SPLINE":
+            return getShortestPath(mesh, sources, targets, "SPLINE", "PATH")
+        elif self.pathType == "STROKE":
+            return getShortestPath(mesh, sources, targets, "STROKE", "PATH")
```

Comment:
Why do we have this code if the path type won't matter?

Reply:
Sorry. I did not realize that. It is removed.